src/platform/inputs.h

```diff
@@ -45,7 +45,8 @@ namespace mlx
 			inline int getYRel() const noexcept { return _yRel; }
 
 			inline bool isRunning() const noexcept { return !_end; }
-			inline constexpr void finish() noexcept { _end = true; }
+			inline constexpr void setRunning(bool running) noexcept { _end = !running; }
+			inline constexpr void finish() noexcept { setRunning(false); }
```

Comment:
I don't know if there's a proper way to mark something as deprecated in C++, so there you go.